app/components/features/target-spf-item/component.tsx

```diff
@@ -83,18 +93,29 @@ export const TargetSPFItem: React.FC<TargetSPFItemProps> = ({
             <span>100%</span>
           </div>
         </div>
-        <div className="flex flex-col justify-between w-24 px-4 border-l">
+        <div className="flex flex-col justify-between w-24 px-4 border-l border-gray-500">
           <span>{isAllTargets ? 'ALL SPF' : 'SPF'}</span>
           <div className="w-10 mb-6">
             <Input
-              className="px-0 py-1"
+              className="px-0 py-1 rounded"
               theme="dark"
               mode="dashed"
               type="number"
-              value={FPFValue}
+              value={inputFPFValue}
               onChange={({ target: { value: inputValue } }) => {
-                setFPFValue(Number(inputValue));
-                if (onChangeFPF) onChangeFPF(Number(inputValue));
+                setInputFPFValue(inputValue);
+              }}
+              onBlur={() => {
+                // If user leaves the input empty, we'll revert to the original targetValue
+                if (!inputFPFValue) {
+                  setInputFPFValue(String(FPFValue));
+                  return;
+                }
+                // Prevent changing all targets if user didn't actually change it
+                // (despite clicking on the input)
+                if (FPFValue === Number(inputFPFValue)) return;
+                setFPFValue(Number(inputFPFValue));
+                if (onChangeFPF) onChangeFPF(Number(inputFPFValue));
```

Comment:
Not sure if last condition is needed

Reply:
It makes it so that, when the Item in question is the "All targets" one, the callback is called to trigger a change on all the items' FPFs.